backend/corpora/common/upload.py

```diff
@@ -52,9 +52,26 @@ def upload(
     scope: str = None,
     dataset_id: str = None,
 ) -> str:
+    # Check if a dataset already exists
+    if dataset_id:
+        dataset = Dataset.get(db_session, dataset_id, collection_id=collection_id)
+        if not dataset:
+            raise NonExistentDatasetException(f"Dataset {dataset_id} does not exist")
+    else:
+        dataset = None
+
+    # Check if file size is within max size limit
     max_file_size_gb = CorporaConfig().upload_max_file_size_gb * GB
     if file_size is not None and file_size > max_file_size_gb:
-        raise MaxFileSizeExceededException(f"{url} exceeds the maximum allowed file size of {max_file_size_gb} Gb")
```

Comment:
`url` is not needed, and I don't want to have to think through the implications of saving Collection / Dataset id's here (probably not an issue?) _as well as_ exposing the curator's google auth id (possibly an issue?)